README.md

```diff
@@ -1,80 +1,90 @@
+```
 
-## {relation_name}.ca.available
-The Certificate Authority is available on the relation object when the 
-"{relation_name}.ca.available" state is set. The charm layer can retrieve the
-CA by calling `get_ca()` method on the relationship object.
+Next, you must ensure the interface layer is included in your `layer.yaml`:
 
-```python
-from charms.reactive import when
-@when('certificates.ca.available')
-def store_ca(tls):
-    certificate_authority = tls.get_ca()
+```yaml
+includes:
+  - interface:tls-certificates
 ```
 
-## {relation_name}.server.cert.available
-Once the server certificate is set on the relation the interface layer will
-emit the "{relation_name}.server.cert.available" state, indicating that the 
-server certificate is available from the relationship object.  The charm layer 
-can retrieve the certificate and use it in the code by calling the
-`get_server_cert()` method on the relationship object.
+Then, in your reactive code, add the following, changing `update_certs` to
+handle the certificates however your charm needs:
 
 ```python
-from charms.reactive import when
-@when('certificates.server.cert.available')
-def get_server(tls):
-    server_cert, server_key = tls.get_server_cert()
-```
+from charmhelpers.core import hookenv, host
+from charms.reactive import endpoint_from_flag
 
-## {relation_name}.client.cert.available
-Once the client certificate is set on the relation the interface layer will
-emit the "{relation_name}.client.cert.available" state, indicated that the
-server certificates is available from the relationship object.  The charm layer
-can retrieve the certificate and use it in the code by calling the
-`get_client_cert()` method on the relationship object.
 
-```python
-from charms.reactive import when
-@when('certificates.client.cert.available')
-def store_client(tls):
-    client_cert, client_key = tls.get_client_cert()
+@when('cert-provider.ca.changed')
+def install_root_ca_cert():
+    cert_provider = endpoint_from_flag('cert-provider.ca.available')
+    host.install_ca_cert(cert_provider.root_ca_cert)
+    clear_flag('cert-provider.ca.changed')
+
+
+@when('cert-provider.available')
+def request_certificates():
+    cert_provider = endpoint_from_flag('cert-provider.available')
+
+    # get ingress info
+    ingress_for_clients = hookenv.network_get('clients')['ingress-addresses']
+    ingress_for_db = hookenv.network_get('db')['ingress-addresses']
+
+    # use first ingress address as primary and any additional as SANs
+    server_cn, server_sans = ingress_for_clients[0], ingress_for_clients[:1]
+    client_cn, client_sans = ingress_for_db[0], ingress_for_db[:1]
+
+    # request a single server and single client cert; note that multiple certs
+    # of either type can be requested as long as they have unique common names
+    cert_provider.request_server_cert(server_cn, server_sans)
+    cert_provider.request_client_cert(client_cn, client_sans)
+
+
+@when('certificates.certs.changed')
```

Comment:
should this (and the one at the end of the handler) be `s/certificates/cert-provider/`?